src/services/facilityService.js

```diff
@@ -0,0 +1,88 @@
+import db from '../models';
+import { Helpers } from '../utils';
+
+const {
+  Facility, sequelize, RoomCategory, Room, FacilityAmenity, Amenity
+} = db;
+const { extractArrayRecords, updateCollection } = Helpers;
+
+/**
+ * A collection of methods that handles the database interactions
+ * for managing a Facility as an entity of the App.
+ *
+ * @class FacilityService
+ */
+class FacilityService {
+  /**
+   * Checks the room category value and creates a new record for any category added
+   * by user.
+   * @static
+   * @param {array} rooms - An array of rooms of a specific facility.
+   * @returns {Promise<Array>} A promise object with updated room properties.
+   * @memberof FacilityService
+   */
+  static async sortRoomCategory(rooms) {
+    const updatedRooms = rooms.map(async (room) => {
+      const { label } = await RoomCategory.findByPk(room.roomCategoryId);
+      if (label === 'Others') {
+        const { id } = await RoomCategory.create({ description: ' ', label: room.newCategory });
```

Comment:
Could have been destructured from RoomCategory
```suggestion
        const { id } = await create({ description: ' ', label: room.newCategory });
```

Reply:
The comment above applies.